hcache/tdb.c

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * TDB backend for the header cache
+ *
+ * @authors
+ * Copyright (C) 2020 Tino Reichardt <milky-neomutt@mcmilk.de>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @page hc_tdb TDB
+ *
+ * Use a GNU dbm file as a header cache backend.
+ */
+
+#include "config.h"
+#include <stddef.h>
+#include "mutt/lib.h"
+#include "backend.h"
+#include "globals.h"
+
+#include <fcntl.h>
+#include <tdb.h>
+
+/**
+ * hcache_tdb_open - Implements HcacheOps::open()
+ */
+static void *hcache_tdb_open(const char *path)
+{
+  int flags = TDB_NOLOCK|TDB_INCOMPATIBLE_HASH|TDB_NOSYNC;
+  return tdb_open(path, 33657, flags, O_CREAT|O_RDWR, 00600);
```

Comment:
Can you provide some background on te choice of flags (especially the undocumented TDB_INCOMPATIBLE_HASH) and of hash size? Perhaps as a comment in the code.

Reply:
- TDB_INCOMPATIBLE_HASH is a lot faster, and we have no other TDB's currently
- the number was some try+fail ... until the speed on my test was nice